cmd/build/v2/services.go

```diff
@@ -74,55 +92,56 @@ func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model
 	return svcsToBuildList, nil
 }
 
-func (bc *OktetoBuilder) checkServicesToBuild(service string, manifest *model.Manifest, ch chan string) error {
+func (bc *OktetoBuilder) getDigestFromService(service string, manifest *model.Manifest) (string, error) {
 	buildInfo := manifest.Build[service].Copy()
-	isStack := manifest.Type == model.StackType
-	if isStack && okteto.IsOkteto() && !registry.IsOktetoRegistry(buildInfo.Image) {
-		buildInfo.Image = ""
-	}
-	tag := getToBuildTag(manifest.Name, service, buildInfo)
-	if tag == "" {
-		return fmt.Errorf("error getting the image name for the service '%s'. Please specify the full name of the image when using a cluster that doesn't have Okteto installed", service)
+
+	tags := getToBuildTags(manifest.Name, service, buildInfo)
+	if len(tags) == 0 {
+		return "", fmt.Errorf("error getting the image name for the service '%s'. Please specify the full name of the image when using a cluster that doesn't have Okteto installed", service)
 	}
 
-	imageWithDigest, err := bc.Registry.GetImageTagWithDigest(tag)
-	if oktetoErrors.IsNotFound(err) {
-		oktetoLog.Debug("image not found, building image")
-		ch <- service
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("error checking image at registry %s: %v", tag, err)
+	for _, tag := range tags {
+		imageWithDigest, err := bc.Registry.GetImageTagWithDigest(tag)
+		if oktetoErrors.IsNotFound(err) {
+			oktetoLog.Debugf("image not found for tag %s, building image", tag)
+			continue
+		} else if err != nil {
+			return "", fmt.Errorf("error checking image at registry %s: %w", tag, err)
+		}
+		return imageWithDigest, nil
```

Comment:
why we are looping over tags but we only return one digest? not sure if I understand this logic

Reply:
We need to check if the image is already built in one of the okteto registries. If found we return that digest, which is the one we'll be using for the image

Reply:
could we looking for that digest for the specific tag? why we need to loop over the tags? for me it is like "I will try if this exists, if not, will try next one..?" Is it?

Reply:
We wanted to point `jlopezbarb/vote`, `okteto.global/vote` and `okteto.dev/vote to the same digest. That's why we try with the three different tags